cluster-api-state-metrics-adoption/README.md

```diff
@@ -0,0 +1,237 @@
+# Cluster API state metrics
+
+<!-- toc -->
+- [Glossary](#glossary)
+- [Problem statement](#problem-statement)
+- [Cluster API state metrics](#cluster-api-state-metrics)
+  - [Infrastructure provider specific metrics](#infrastructure-provider-specific-metrics)
+    - [all providers in one binary](#all-providers-in-one-binary)
+    - [code separation per provider](#code-separation-per-provider)
+    - [Conclusion](#conclusion)
+- [Tasks and Stories](#tasks-and-stories)
+  - [Story 1](#story-1)
+  - [Story 2](#story-2)
+  - [Story 3](#story-3)
+<!-- /toc -->
+
+This RFC motivates the adoption of the existing Cluster API state metrics project into our own workflow.
+
+## Glossary
+
+- Infrastructure providers - Cluster API infrastructure implementation for e.g. AWS (CAPA), OpenStack (CAPO), Google Compute Platorm (CAPG), ...
+- `cluster-api-state-metrics` (CASM)
+- Management Cluster (MC) is the Kubernetes Cluster where CAPI components are deployed and which is responsible for creating new Kubernetes Clusters (WCs)
+- Workload Cluster (WC)
+
+## Problem statement
+
+As Cluster Operator we want to get know the state of any Cluster API related object (no matter if Cluster API Core or infrastructure provider specific).
+
+As different states have impact on the expected behavior of the reconciliation e.g.
+
+- A cluster upgrade won't finish successful if an already existing OpenStackMachine is in an error state (even if the machine is in a transient error).
```

Comment:
Note that we have MachineHealthCheck and it fixes transient errors.

Reply:
> Note that we have MachineHealthCheck and it fixes transient errors.

unfortunately no, as the Node itself is up and running and healthy. Only the state/status of the machine might be in a failure state due a transient error with e.g. the Openstack API - API temp. not available.

Reply:
When `OpenStackMachine` has an error, isn't it propagated to the corresponding `Machine` CR?

Reply:
no, doesn't propagated to the `Machine` (xref: https://github.com/giantswarm/giantswarm/issues/22077)